src/types/index.ts

```diff
@@ -890,3 +891,18 @@ export interface IAmountDecimalPlaces {
   highPrecision?: boolean;
   amount?: number;
 }
+
+export interface IAddressBookEntry {
+  id: string;
+  name: string;
+  address: AccountAddress;
+  isBookmarked: boolean;
+  protocol: Protocol;
+}
+
+/**
+ * The `string` stands for the id of the entry.
+ */
+export type IAddressBook = Record<string, IAddressBookEntry>;
```

Comment:
After spending some time with the PR and having a conversation on the chat I'm sure that using random generated IDs for keying the entries is rather bad (superfluous). Take a look on how the network management is done.

I think the address or eventually the index is enough to allow to distinguish the entries between each other.

So the entries could be stored as an array instead of object.